Movie Challenge/Movie Challenge/Controllers/Detail/DetailViewController.swift

```diff
@@ -0,0 +1,153 @@
+            self.posterImageView.image = poster
+            self.posterImageView.setNeedsDisplay()
+        }
+    }
+    
+    private func setButtonState(){
+        if(favorite){
+            DispatchQueue.main.async(){
+                self.favoriteButtonView.backgroundColor = #colorLiteral(red: 0.9254902005, green: 0.2352941185, blue: 0.1019607857, alpha: 1)
+                self.favoriteButtonView.setTitle("Remover", for: UIControlState.normal)
+            }
+        }else{
+            DispatchQueue.main.async(){
+                self.favoriteButtonView.backgroundColor = #colorLiteral(red: 1, green: 0.6116010603, blue: 0.006196474039, alpha: 1)
+                self.favoriteButtonView.setTitle("Favoritar", for: UIControlState.normal)
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        categoryCollectionView.delegate = self
+        categoryCollectionView.dataSource = self
+        
+        self.movieId = movieId ?? 76341
+        
+        do{
+            self.movie = try MovieRepository.shared().getOne(by: movieId)
+            self.favorite = true
+            self.setButtonState()
+            self.fillFields()
+        }catch{
+            self.favorite = false
+            
+            _ = MovieService.shared().getMovieDetail(id: movieId){ movie, response, error in
+                self.movie = movie
+                self.setButtonState()
+                self.fillFields()
+            }
+        }
+    }
+    
+    public func setUp(movieId: Int?){
+        //TODO MAD MAX CODE
+        self.movieId = movieId ?? 76341
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let genres = movie.genres else { return 0 }
+        
+        return genres.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "CategoryCell", for: indexPath) as! CategoryCollectionViewCell
+        
+        guard let categoryList = movie.genres else{ return UICollectionViewCell() }
+        cell.setUp(name: categoryList[indexPath.row].name)
+        
+        return cell
+    }
+    
+    func fillFields(){
+        DispatchQueue.main.async(){
+            
+            self.titleLabelView.text = self.movie.title
+            self.pointsLabelView.text = String(self.movie.vote_average!)
+            self.numberOfVotesLabelView.text = "(" + String(self.movie.vote_count!) + ")"
+            self.overviewLabelView.text = self.movie.overview
+            
+            if let releaseDate = self.movie.release_date{
+                if(!releaseDate.isEmpty){
+                    self.yearLabelView.text = String((self.movie.release_date?.split(separator: "-").first)!)
+                }
+            }
+            
+            if let runtime = self.movie.runtime{
```

Comment:
I personally like of a space between the sentence and the "{", like this:
if myVariable {
}. But this is a optional request, do if you want.